File: src/ast/NodeArray.java
package ast;

import java.io.StringWriter;
import java.util.LinkedList;
import java.util.List;

public class NodeArray extends BaseNode {
    public List<BaseNode> Nodes;

    public NodeArray(List<BaseNode> nodes) {
        super(NodeType.NodeArray);
        Nodes = nodes;
    }

    public NodeArray(List<BaseNode> nodes, NodeType type) {
        super(type);
        Nodes = nodes;
    }

    @Override
    public boolean IsArray() {
        return true;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        List<String> nodeStrings = new LinkedList<>();
        for(BaseNode node: Nodes){
            nodeStrings.add(node.toString());
        }
        String[] nodeStringsArr = nodeStrings.toArray(new String[0]);
        writer.write(String.join(", ", nodeStringsArr));
    }
}

File: src/de/fearlesstobi/demangler/ast/NodeArray.kt
package de.fearlesstobi.demangler.ast

import de.fearlesstobi.demangler.util.StringUtil
import java.io.StringWriter

open class NodeArray : BaseNode {
    val nodes: List<BaseNode?>?

    constructor(nodes: List<BaseNode?>?) : super(NodeType.NodeArray) {
        this.nodes = nodes
    }

    constructor(nodes: List<BaseNode?>?, type: NodeType) : super(type) {
        this.nodes = nodes
    }

    override val isArray: Boolean
        get() = true

    public override fun printLeft(writer: StringWriter) {
        writer.write(StringUtil.nodeListToArray(nodes).joinToString { ", " })
    }
}